### PR TITLE
New approximation methods: KNeighbors and RadiusNeighbors

### PR DESCRIPTION
**Pull request by fAndreuzzi**
This commit adds two new approximation methods: `KNeighborsRegressor` and `RadiusNeighborsRegressor`. They are wrappers for the _sklearn_ classes [KNeighborsRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsRegressor.html#sklearn.neighbors.KNeighborsRegressor) and [RadiusNeighborsRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.RadiusNeighborsRegressor.html#sklearn.neighbors.RadiusNeighborsRegressor). 

The two methods are tested in isolation and together with the _EZyRB_ interface.

### CODE REVIEW

File: ezyrb/neighborsregressor.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class NeighborsRegressor(Approximation):
+    """
+    A generic superclass for wrappers of *NeighborsRegressor from sklearn.
+
+    :param kwargs: arguments passed to the internal instance of
+        *NeighborsRegressor.
+    """
+
+    def __init__(self, **kwargs):
+        raise ValueError("This class must be extended, not used.")
```
Comment (mtezzele) [inline ezyrb/neighborsregressor.py:14]: Please check that this is the correct error: https://docs.python.org/3/library/exceptions.html

File: tests/test_radius_neighbors_regressor.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        assert rom.predict([2]) == 5
+        assert rom.predict([3]) == 3
+
+    def test_wrong(self):
+        # wrong number of params
+        with self.assertRaises(Exception):
+            reg = RadiusNeighborsRegressor()
+            reg.fit([[1, 2], [6,], [8, 9]], [[1, 0], [20, 5], [8, 6]])
+
+        # wrong number of values
```
Comment (mtezzele) [inline tests/test_radius_neighbors_regressor.py:65]: Please split tests

File: tests/test_k_neighbors_regressor.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        assert rom.predict([2]) == 5
+        assert rom.predict([3]) == 3
+
+    def test_wrong(self):
+        # wrong number of params
+        with self.assertRaises(Exception):
+            reg = KNeighborsRegressor()
+            reg.fit([[1, 2], [6,], [8, 9]], [[1, 0], [20, 5], [8, 6]])
+
+        # wrong number of values
```
Comment (mtezzele) [inline tests/test_k_neighbors_regressor.py:65]: Please split

**Review by mtezzele — COMMENTED** [review]
Thanks! I think you should extend the list `__all__` in the `__init__.py` with the new modules.

**Comment by fAndreuzzi** [comment]
In my last commit I should have applied all the improvements you requested, @mtezzele.

**Comment by fAndreuzzi** [comment]
I updated `__all__` and fixed some mispelled names in my last commit.

**Pull request merged**